pkg/scheduler/algorithm/predicates/error.go

```diff
@@ -20,46 +20,20 @@ import (
 	"fmt"
 
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodeaffinity"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodename"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodeports"
 )
 
 var (
-	// The predicateName tries to be consistent as the predicate name used in DefaultAlgorithmProvider defined in
-	// defaults.go (which tend to be stable for backward compatibility)
-
-	// NOTE: If you add a new predicate failure error for a predicate that can never
-	// be made to pass by removing pods, or you change an existing predicate so that
-	// it can never be made to pass by removing pods, you need to add the predicate
-	// failure error in nodesWherePreemptionMightHelp() in scheduler/core/generic_scheduler.go
-
 	// ErrNodeSelectorNotMatch is used for MatchNodeSelector predicate error.
-	ErrNodeSelectorNotMatch = NewPredicateFailureError("MatchNodeSelector", "node(s) didn't match node selector")
+	ErrNodeSelectorNotMatch = NewPredicateFailureError(nodeaffinity.Name, nodeaffinity.ErrReason)
```

Comment:
Should we update `pkg/scheduler/core/generic_scheduler_test.go` to import error from `nodeaffinity` or could keep 
```
			nodesStatuses: framework.NodeToStatusMap{
				"machine1": framework.NewStatus(framework.UnschedulableAndUnresolvable, algorithmpredicates.ErrNodeSelectorNotMatch.GetReason()),
				"machine2": framework.NewStatus(framework.UnschedulableAndUnresolvable, algorithmpredicates.ErrPodNotMatchHostName.GetReason()),

```